config/v1/types_oauth.go

```diff
@@ -1,37 +1,514 @@
+	// AccessTokenInactivityTimeoutSeconds defines the default token
+	// inactivity timeout for tokens granted by any client.
+	// The value represents the maximum amount of time that can occur between
+	// consecutive uses of the token. Tokens become invalid if they are not
+	// used within this temporal window. The user will need to acquire a new
+	// token to regain access once a token times out.
+	// Valid values are integer values:
+	//   x < 0  Tokens never timeout (e.g. `-1`)
+	//   x = 0  Tokens are disabled (default)
+	//   x > 0  Tokens time out if there is no activity for x seconds
+	// The current minimum allowed value for X is 300 (5 minutes)
+	// +optional
+	AccessTokenInactivityTimeoutSeconds int32 `json:"accessTokenInactivityTimeoutSeconds,omitempty"`
+}
+
+const (
+	// LoginTemplateKey is the default key of the login template
+	LoginTemplateKey = "login.html"
+	// ProviderSelectionTemplateKey is the default key for the provider selection template
+	ProviderSelectionTemplateKey = "providers.html"
+	// ErrorsTemplateKey is the default key for the errors template
+	ErrorsTemplateKey = "errors.html"
+)
+
+// OAuthTemplates allow for customization of pages like the login page
+type OAuthTemplates struct {
+	// Login is a reference to a secret that specifies a go template to use to render the login page.
+	// If a key is not specified, the key `login.html` is used to locate the template data.
+	// If unspecified, the default login page is used.
+	// +optional
+	Login LocalSecretReference `json:"login,omitemtpy"`
+
+	// ProviderSelection is a reference to a secret that specifies a go template to use to render
+	// the provider selection page.
+	// If a key is not specified, the key `providers.html` is used to locate the template data.
+	// If unspecified, the default provider selection page is used.
+	// +optional
+	ProviderSelection LocalSecretReference `json:"providerSelection,omitempty"`
+
+	// Error is a reference to a secret that specifies a go template to use to render error pages
+	// during the authentication or grant flow.
+	// If a key is not specified, the key `errrors.html` is used to locate the template data.
+	// If unspecified, the default error page is used.
+	// +optional
+	Error LocalSecretReference `json:"error,omitempty"`
+}
+
+// IdentityProvider provides identities for users authenticating using credentials
+type IdentityProvider struct {
+	// Name is used to qualify the identities returned by this provider.
+	// - It MUST be unique and not shared by any other identity provider used
+	// - It MUST be a vlid path segment: name cannot equal "." or ".." or contain "/" or "%"
+	//   Ref: https://godoc.org/k8s.io/apimachinery/pkg/api/validation/path#ValidatePathSegmentName
+	Name string `json:"name"`
+
+	// DisplayName is a user-friendly name to display in provider selection options.
+	// If not provided, the `name` will be used instead.
+	// +optional
+	DisplayName string `json:"displayName"`
```

Comment:
This one came up on the sme mailing list recently as a desired feature. Figured if we want to add something like it, we might as well do it now. cc @enj

Reply:
Please remove this.  It can already be done via the go template for the selection provider.

Reply:
I really don't want to encourage using those templates. I'll remove for now since we can always add it later.

Reply:
The templates exist to allow uses to brand and otherwise customize the look and feel of the OAuth pages (including things like friendly names).  Even the online clusters require this to limit user confusion on internal vs external IDPs.  We will never be able to remove them.